### PR TITLE
[FEATURE][rasters] Invert colors' raster rendering filter option

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR adds a nifty [x] invert colors setting to raster rendering options. This can come in handy, for e.g. when needing a dark base map using out-of-the-box OSM tiles:

https://user-images.githubusercontent.com/1728657/122640364-5c757d00-d129-11eb-94b1-d7f81ae950c4.mp4

### CODE REVIEW

File: src/core/raster/qgshuesaturationfilter.cpp [hunk 1]
```diff
@@ -357,6 +369,7 @@ void QgsHueSaturationFilter::readXml( const QDomElement &filterElem )
 
   setSaturation( filterElem.attribute( QStringLiteral( "saturation" ), QStringLiteral( "0" ) ).toInt() );
   mGrayscaleMode = ( QgsHueSaturationFilter::GrayscaleMode )filterElem.attribute( QStringLiteral( "grayscaleMode" ), QStringLiteral( "0" ) ).toInt();
+  mInvertColors = ( bool )filterElem.attribute( QStringLiteral( "invertColors" ), QStringLiteral( "0" ) ).toInt();
```
Comment (nyalldawson) [inline src/core/raster/qgshuesaturationfilter.cpp:372]: Static cast please

File: src/core/raster/qgshuesaturationfilter.h [hunk 2]
```diff
@@ -56,6 +56,11 @@ class CORE_EXPORT QgsHueSaturationFilter : public QgsRasterInterface
     void setSaturation( int saturation );
     int saturation() const { return mSaturation; }
 
+    //! Sets whether the filter will invert colors.
+    void setInvertColors( bool invertColors ) { mInvertColors = invertColors; }
```
Comment (nyalldawson) [inline src/core/raster/qgshuesaturationfilter.h:60]: Since

**Review by nyalldawson — COMMENTED** [review]
Code looks good -- can you add tests too?

**Comment by nirvn** [comment]
@nyalldawson , comments addressed.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**